src/types/profile.js

```diff
@@ -223,3 +223,11 @@ export type Profile = {
   tasktracer: TaskTracer,
   threads: Thread[],
 };
+
+export type ProfileSharingCompositeButtonProps = {
+  profile: Object,
+  dataSource: string,
+  hash: string,
+  onProfilePublished: Function,
+  predictURL: Function,
+};
```

Comment:
You can define this props object directly in the previous file, it's not necessary to extract it. See the previous examples for some guidance :)